modules/core/src/lib/effect-manager.js

```diff
@@ -44,23 +39,40 @@ export default class EffectManager {
   setEffects(effects = []) {
     this.cleanup();
     this.effects = effects;
+    this._createInternalEffects();
   }
 
   cleanup() {
     for (const effect of this.effects) {
       effect.cleanup();
     }
+
+    for (const effect of this._internalEffects) {
+      effect.cleanup();
+    }
     this.effects.length = 0;
+    this._internalEffects.length = 0;
   }
 
-  checkLightingEffect() {
+  _needApplyLightingEffect() {
```

Comment:
Is it necessary to create a separate method for this? Where else is it needed?

Reply:
Just make the code structure cleaner with more functions to wrap specific logics